iocore/net/quic/QUICStream.cc

```diff
@@ -408,7 +408,26 @@ QUICStream::generate_frame(QUICEncryptionLevel level, uint64_t connection_credit
 {
   SCOPED_MUTEX_LOCK(lock, this->_write_vio.mutex, this_ethread());
 
-  QUICFrameUPtr frame = QUICFrameFactory::create_null_frame();
+  // retransmitted frame has higher priority.
+  QUICFrameUPtr frame = this->pop_from_retransmitted_queue();
+  if (frame != nullptr && this->_state.is_allowed_to_send(*frame)) {
+    // once we are in the terminal state, the state wouldn't change. So the frame would be prohibited forever
+    // So we just skip this type of frame and try to generate a new type of frame.
+    if (frame->size() > maximum_frame_size) {
+      auto new_frame = QUICFrameFactory::split_frame(frame.get(), maximum_frame_size);
```

Comment:
This would work, but I'm not so keen on this way (splitting frames).

Do we need to create frames in `_on_frame_lost`? If we could create frames that fit in maximum_frame_size here instead, we wouldn't need to split frames.

Reply:
In _on_frame_lost, we don't know the real value of `maximum_frame_size` . So how to split it into an appropriate limited size.

Reply:
No, I'm suggesting that push a frame-information (`StreamFrameInfo`) into `lost_info_queue` in `_on_frame_lost`, and then pop it and create a frame in `generate_frame`.

Reply:
got it.